File: program/convert-octal-to-hexadecimal/ConvertOctalToHexadecimal.ts
function octalToDecimal(n:any){
    const num = n;
    let dec_value = 0;
    let base = 1;
   
    let temp = num;
    while (temp > 0){
        const last_digit = temp % 10;
        temp = Math.floor(temp / 10);
        dec_value += last_digit * base;
   
        base = base * 8;
    }
    return dec_value;
}

function decToHexa(n:any){
    const hexaDeciNum = new Array(100);
    let i = 0;
    while(n != 0){   
        let temp  = 0;
        temp = n % 16;
        
        if (temp < 10){
            hexaDeciNum[i] = temp + 48;
            i++;
        }
        else{
            hexaDeciNum[i] = temp + 87;
            i++;
        }
        n = Math.floor(n / 16);
    }
    let ans = "";
    for(let j = i - 1; j >= 0; j--){
        ans += String.fromCharCode(hexaDeciNum[j]);
    }
    return ans;
}

let hexnum;
let decnum, octnum;

octnum = 12;
     
// Convert Octal to Decimal
decnum = octalToDecimal(octnum);
     
// Convert Decimal to Hexadecimal
hexnum = decToHexa(decnum);
 
console.log("Equivalent Hexadecimal Value :" + hexnum);
 

File: program/convert-octal-to-hexadecimal/ConvertOctalToHexadecimal.cs
using System;
class Conversion{
static int octalToDecimal(int n)
{
int num = n;
int dec_value = 0;

int b_ase = 1;

int temp = num;
while (temp > 0)
{
	int last_digit = temp % 10;
	temp = temp / 10;

	dec_value += last_digit * b_ase;

	b_ase = b_ase * 8;
}

return dec_value;
}

static string decToHexa(int n)
{
char []hexaDeciNum = new char[100];

int i = 0;
string ans = "";

while(n != 0)
{
	int temp = 0;
	temp = n % 16;
	if(temp < 10)
	{
	hexaDeciNum[i] = (char)(temp + 48);
	i++;
	}
	else
	{
	hexaDeciNum[i] = (char)(temp + 87);
	i++;
	}

	n = n / 16;
}

for(int j = i - 1; j >= 0; j--)
{
	ans += hexaDeciNum[j];
}

return ans;
}

public static void Main(string []args)
{
string octnum, hexnum;
int decnum;

Console.WriteLine ("Please enter an Octal number :");
octnum = Console.ReadLine();

decnum = octalToDecimal(Int32.Parse(octnum));

hexnum = decToHexa(decnum);

Console.Write("Equivalent Hexadecimal Value = " +
				hexnum);
}
}

